### ISSUE TITLE
Question regarding dependency

### ISSUE THREAD

**Issue by s1mark** [gruntwork-io/terragrunt#1672]
### Scenario
* `module_a` depends `module_c`
* `module_b` depends `module_c`

### Question
Will `module_c` be invoked twice in case running `terragrunt run-all apply` ?

The same question would be for the `vpc` in the https://terragrunt.gruntwork.io/docs/features/execute-terraform-commands-on-multiple-modules-at-once/#dependencies-between-modules example

**Comment by s1mark** [comment]
The same question would be for the `vpc` in the https://terragrunt.gruntwork.io/docs/features/execute-terraform-commands-on-multiple-modules-at-once/#dependencies-between-modules example

**Comment by nkaravias** [comment]
Aren't modules supposed to be idempotent (in which case it doesn't matter if its invoked twice or not)?

**Comment by brikis98** [comment]
No, Terragrunt runs each module only once.

**Comment by yorinasub17** [comment]
Closing this as the original question has been answered. If there are any follow ups, please open a new ticket with updated context. Thanks!
